update-go-version.sh

```diff
@@ -15,6 +15,6 @@ sed -i "s,^\(FROM registry.ci.openshift.org/openshift/release:rhel-8-release-gol
 sed -i "s,^FROM registry.access.redhat.com/ubi8/go-toolset:[.0-9]\+,FROM registry.access.redhat.com/ubi8/go-toolset:${golang_base_version}," images/*/Dockerfile
 for f in .github/workflows/*.yml; do
     if [ $(yq '.jobs.build.strategy.matrix | has("go")' "$f") == "true" ]; then
-      yq eval --inplace ".jobs.build.strategy.matrix.go[0] = ${golang_base_version}" "$f";
+      yq eval --inplace ".jobs.build.strategy.matrix.go[0] = ${golang_base_version} | .jobs.build.strategy.matrix.go[0] style=\"single\"" "$f";
```

Comment:
I'm not sure what this is fixing? All the workflows file have single quotes around the go version, and if I run `update-go-version 1.21` without this commit, then I get quoted `'1.21'` in the workflows  files.

Reply:
I ran more tests, and this is needed because otherwise 
```
        go:
         - '1.20'
```
gets replaced with
```
        go:
          - 1.21
```

But this behaviour only started happening in yq 4.21.1, I don't know if this was intentional.

Reply:
I don't know, was reading the doc and because I was not getting single quotes and found out the style operator to use.

Reply:
Might be nice to detail the commit log a bit more
```
 is not the case till now.
```
-> "which is not the case with yq 4.21.1 and newer. `yq eval --inplace ".jobs.build.strategy.matrix.go[0] = 1.21` replaces `go: '1.20'` with `go: 1.21` without quotes"